src/stream/stream/mod.rs

```diff
@@ -241,6 +243,35 @@ extension_trait! {
             }
         }
 
+        #[doc = r#"
+            Creates a stream that yields elements based on a predicate.
+
+            # Examples
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use std::collections::VecDeque;
+
+            use async_std::prelude::*;
+
+            let s: VecDeque<usize> = vec![1, 2, 3, 4].into_iter().collect();
+            let mut s = s.take_while(|x| x < &3 );
+
+            assert_eq!(s.next().await, Some(1));
+            assert_eq!(s.next().await, Some(2));
+            assert_eq!(s.next().await, None);
+
+            #
+            # }) }
+        "#]
+        fn take_while<P>(self, predicate: P) -> TakeWhile<Self, P, Self::Item>
+            where
+                Self: Sized,
+                P: FnMut(&Self::Item) -> bool,
```

Comment:
nit: can we unindent these three lines to keep the same code formatting as in other methods? :)

Reply:
sure